### ISSUE TITLE
when Chinese Input Method input char not match pattern, ng-model brokened

### ISSUE THREAD

**Issue by lovemyliwu** [AlphaGit/ng-pattern-restrict#42]
```
<input maxlength="6"
                                           type="text"
                                           ng-model="emailVerifyCode"
                                           pattern="(^$|^[0-9]+$)"
                                           ng-pattern-restrict
                                           class="text-input verification-code"
                                    />
```

after use Chinese Input Method input Chinese Char, ng-model directive broken.

**Comment by lovemyliwu** [comment]
as a hotfix I tried to manually apply new value at https://github.com/AlphaGit/ng-pattern-restrict/blob/master/src/ng-pattern-restrict.js#L140

```
         function updateCurrentValue(newValue) {
            oldValue = newValue;
            caretPosition = getCaretPosition();

           if (ngModelController) {
              scope.$apply(function () {
                ngModelController.$setViewValue(oldValue);
              });
            }
          }
```

**Comment by lovemyliwu** [comment]
i create a PR #47

**Comment by AlphaGit** [comment]
Fixed in #44.
